src/commands/setup/validation.ts

```diff
@@ -0,0 +1,42 @@
+import { ApiKeyNotProvided, DatadogSiteValueError } from "./errors";
+import { DatadogSite, datadogSites } from "./interface";
+import { SetupCommandInitialStateType, SetupCommandStateType } from "./setup";
+
+const isDatadogSite = (value: unknown): value is DatadogSite => {
+  return datadogSites.includes(value as DatadogSite);
+};
+
+export const validateInitialState = (
+  initialState: SetupCommandInitialStateType
+): SetupCommandStateType => {
+  if (initialState.bypassPrompts && !initialState.apiKey) {
+    throw new ApiKeyNotProvided(initialState.apiKey || "");
```

Comment:
if I get it right, `initialState.apiKey` should be null, or empty, or undefined in this branch. Is it useful to include one of these values into error?

Reply:
Yes I can use `JSON.stringify` here, should give more data